WordPress/gutenberg
# Buttons: Clean up editor styles and fix button text transform

**🚧  Do not merge before #56649 🚧** 

Depends on: 
- https://github.com/WordPress/gutenberg/pull/56649

Related:
- https://github.com/WordPress/gutenberg/pull/43934#discussion_r1408015884

## What?

- Removes unnecessary editor styles for the Buttons and Button blocks. 
- Fixes inheritance of text transform on a button instance to overcome global styles correctly

## Why?

- Reduced CSS is always good.
- Seeing expected styling is good too.

## How?

- Delete the bad CSS
- Add a little necessary CSS

## Testing Instructions

1. Edit a post and add a few buttons
2. Confirm that setting a border color only, on an individual button, applied the `border-width: initial` CSS rule and a border displays
3. Repeat the process for each individual side i.e. top, right, bottom, left
4. Reset the border styles for the button and repeat again setting a border style
5. Apply a text transform style to the button and ensure that the inner `.wp-block-button__link` element gets `text-transform: inherit` and displays the correct transform
6. Reset individual button styles and select the parent Buttons block
7. Apply a font size on the parent block and confirm it gets applied to the inner Button blocks correctly
